docs/13-trino.md

```diff
@@ -0,0 +1,474 @@
+
+### Features
+
+This plugin supports the following SQL features:
+
+- Sleeper tables and their columns are exposed as SQL tables in Trino
+- Single-value filters, such as SELECT * FROM sleepertable WHERE rowkey = 'key1'
+- Multiple single-value filters, such as SELECT * FROM sleepertable WHERE rowkey IN ('key1', 'key10', 'key100')
+- Range queries, such as SELECT * FROM sleepertable WHERE rowkey BETWEEN 'key1' AND 'key10'
+- Pattern-based queries with a prefix, such as SELECT * FROM sleepertable WHERE rowkey LIKE 'key1%'
+- Joins between tables, where the join key extracted from the first table is the same as the row key on the second
+  table, such as SELECT * FROM table1 INNER JOIN table2 USING (table1.rowkey = table2.extractedkey) WHERE table2.rowkey
+  = 'key1'
+- Insert statements, such as INSERT INTO table1 VALUES ('Fred', 27, 'M') for small numbers of rows
+- Insert statements involving many millions of rows, such as INSERT INTO table1 SELECT * FROM remotesource
+- Example procedures to control Sleeper from within SQL
+- Example system tables to provide up-to-date information about Sleeper, such as lists of the current partitions
+
+### Pushdown and partitioning
+
+Filters which appear in the SQL expression are pushed down to Sleeper, so that the minimum number of rows are retrieved
+from the Sleeper data store. Unfiltered full-table-scans, where there is no rowkey filter, are not permitted in order to
+protect the system and they cause an error.
+
+All other SQL features, such as casting, arithmetic, aggregation and window functions, are supported within the Trino
+execution engine and are not pushed down into the Sleeper connector.
+
+Query scans are distributed and may be executed on multiple servers if this is required. Each Sleeper partition is
+assigned to its own Trino split, and so whenever a query needs to scan more than one Sleeper partition, each split will
+be executed in parallel, wherever possible, by the Trino execution engine.
+
+This plugin supports dynamic filters, which are used to enable efficient joins between tables. The use of dynamic
+filters is affected by the order in which a SQL JOIN expression is written, and this is discussed in more detail below.
+
+Insert statements are partitioned so that the Trino partitions match Sleeper partitions. This ensures that the data for
+any one Sleeper partition is written by a single Trino writer, and this means that the minimum number of files are
+generated within S3. Different Sleeper partitions may be written to by different Trino writers and so writing can take
+place in parallel across multiple servers.
+
+## Installation
+
+### Build the plugin
+
+All of the Sleeper-Trino plugin code is contained in the trino directory. Run Maven in the root directory of this
+package:
+> cd sleeper/java/trino
+>
+> mvn clean package
+
+This will download the required dependencies and build a shaded jar called 'target/sleeper-trino-plugin-XXX.jar'.
```

Comment:
The jar that is generated is called trino-0.11.1-SNAPSHOT-utility.jar - there isn't a jar with 'plugin' in its name.